File: packages/snow_draw_core/lib/draw/render/planning/highlight_mask_visibility.dart
import 'package:snow_draw_core/snow_draw_core.dart';

/// Whether highlight-mask overlay pixels should be painted.
bool isHighlightMaskVisible({
  required bool hasHighlights,
  required HighlightMaskConfig config,
}) => hasHighlights && config.maskOpacity > 0;

File: packages/snow_draw_rust_engine/src/draw/render/planning/highlight_mask_visibility.rs
#![allow(dead_code)]

use crate::draw::config::highlight_config::HighlightMaskConfig;

/// Returns whether highlight-mask pixels should be rendered.
pub fn is_highlight_mask_visible(has_highlights: bool, config: HighlightMaskConfig) -> bool {
    has_highlights && config.mask_opacity > 0.0
}
